File: missing_number.cpp
class Solution {
public:
	int missingNumber(vector<int>& nums) {
		// Your code goes here
    int missing = nums.size();

    for (int i = 0; i < nums.size(); i++) {
        missing ^= i ^ nums[i];
    }

    return missing;
	}
};

File: missing_number.js
class Solution {
	/** 
	 * @param {number[]} nums
	 * @return {number}
	 */
	missingNumber(nums) {
		// Your code goes here
        let n = nums.length;
        let expectedSum = n*(n+1)/2;
        let actualSum = 0;
        nums.forEach((num)=>{
            actualSum += num;
        })
        return expectedSum-actualSum;
	}
};
